tests/e2e/pageobjects/cell.page.ts

```diff
@@ -0,0 +1,138 @@
+import { BasePage } from 'wdio-vscode-service'
+
+export enum OutputType {
+    TerminalView = 'terminal-view',
+    ShellOutput = 'shell-output',
+    Display = '.display'
+}
+
+export interface NotebookCommand {
+    element$: WebdriverIO.Element
+    text: string
+}
+
+export class Cell extends BasePage<{}, { cell: {} }> {
+    public locatorKey = 'cell' as const
+    #cellRow: WebdriverIO.Element
+    #cellText: string
+    constructor(cellRow: WebdriverIO.Element, cellText: string) {
+        // TODO: Write the locator map
+        super({ cell: {} })
+        this.#cellRow = cellRow
+        this.#cellText = cellText
+    }
+
+    async focus() {
+        const container = await this.#cellRow.parentElement().parentElement()
+        await container.click()
+    }
+
+    async run() {
+        const container = await this.#cellRow.parentElement().parentElement()
+        await container.$('.run-button-container').click()
+    }
+
+    async getStatusBarElements(): Promise<NotebookCommand[]> {
+        const successStateRows = await browser.$$('.codicon-notebook-state-success')
+        let commands: NotebookCommand[] = []
+
+        for await (const row of successStateRows) {
+            const parent = await row
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+                .parentElement()
+            const cellEditorContainer = await parent.$('.cell-editor-container')
+            if (await cellEditorContainer.getText() === this.#cellText) {
```

Comment:
It seems like this looks very much like what is done in `isSuccessfulExecution`. Maybe we can have a function like:
```ts
getCurrentExecutionRow () {
    const successRows = await browser.$$('.codicon-notebook-state-success')
    for await (const row of successRows) {
        const executionRow = await row
            .parentElement()
            .parentElement()
            .parentElement()
            .parentElement()
            .parentElement()
        const cellEditor = await executionRow.$('.cell-editor-container')
        const text = await cellEditor.getText()
        if (text === this.#cellText) {
            return executionRow
        }
    }
}
```

and shrink this method down to:

```ts
const commands: NotebookCommand[] = []
const execRow = this.#getCurrentExecutionRow()
if (execRow) {
    return commands
}

const statusBar$ = await execRow.$('.cell-statusbar-container')
const statusRight$ = await statusBar$.$('.cell-status-right')
const contributedRight$ = await statusRight$.$('.cell-contributed-items-right')
const commandsResult$$ = await contributedRight$.$$('.cell-status-item-has-command')

for await (const row of commandsResult$$) {
    const text = await row.getText()
    commands.push({
        element$: row,
        text: text.trim()
    })
}

return commands
```